internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go

```diff
@@ -210,16 +299,194 @@ func newJWTAuthenticator(spec *auth1alpha1.JWTAuthenticatorSpec) (*jwtAuthentica
+			Type:    typeJWKSURLValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidProviderJWKSURL,
+			Message: msg,
+		})
+		// resync err, the user may not be able to fix this via config, it may be the server may be misbehaving.
+		return pJSON.JWKSURL, conditions, fmt.Errorf("%s: %w", errText, err)
+	}
+
+	// spec asserts https is required. https://openid.net/specs/openid-connect-discovery-1_0.html
+	if parsedJWKSURL.Scheme != "https" {
+		msg := fmt.Sprintf("jwks_uri %s has invalid scheme, require 'https'", pJSON.JWKSURL)
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeJWKSURLValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidProviderJWKSURLScheme,
+			Message: msg,
+		})
+		return pJSON.JWKSURL, conditions, fmt.Errorf("%s", msg)
+	}
+
+	conditions = append(conditions, &metav1.Condition{
+		Type:    typeJWKSURLValid,
+		Status:  metav1.ConditionTrue,
+		Reason:  reasonSuccess,
+		Message: "jwks_uri is a valid URL",
+	})
+	return pJSON.JWKSURL, conditions, nil
+}
+
+func (c *jwtCacheFillerController) validateTLS(spec *auth1alpha1.JWTAuthenticatorSpec, conditions []*metav1.Condition) (*x509.CertPool, []*metav1.Condition, bool) {
+	rootCAs, _, err := pinnipedauthenticator.CABundle(spec.TLS)
+	if err != nil {
+		msg := fmt.Sprintf("%s: %s", "invalid TLS configuration", err.Error())
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeTLSConfigurationValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidTLSConfiguration,
+			Message: msg,
+		})
+		return rootCAs, conditions, false
+	}
+
+	msg := "valid TLS configuration"
+	conditions = append(conditions, &metav1.Condition{
+		Type:    typeTLSConfigurationValid,
+		Status:  metav1.ConditionTrue,
+		Reason:  reasonSuccess,
+		Message: msg,
+	})
+	return rootCAs, conditions, true
+}
+
+func (c *jwtCacheFillerController) validateIssuer(issuer string, conditions []*metav1.Condition) (*url.URL, []*metav1.Condition, bool) {
+	issuerURL, err := url.Parse(issuer)
+	if err != nil {
+		msg := fmt.Sprintf("%s: %s", "spec.issuer URL is invalid", err.Error())
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeIssuerURLValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidIssuerURL,
+			Message: msg,
+		})
+		return nil, conditions, false
+	}
+
+	if issuerURL.Scheme != "https" {
+		msg := fmt.Sprintf("spec.issuer %s has invalid scheme, require 'https'", issuer)
+		conditions = append(conditions, &metav1.Condition{
+			Type:    typeIssuerURLValid,
+			Status:  metav1.ConditionFalse,
+			Reason:  reasonInvalidIssuerURLScheme,
+			Message: msg,
+		})
+		return nil, conditions, false
+	}
+
+	conditions = append(conditions, &metav1.Condition{
+		Type:    typeIssuerURLValid,
+		Status:  metav1.ConditionTrue,
+		Reason:  reasonSuccess,
+		Message: "issuer is a valid URL",
+	})
+	return issuerURL, conditions, true
+}
+
+func hadErrorCondition(conditions []*metav1.Condition) bool {
```

Comment:
Refactor opportunity: Looks like this was copied from the federation domains controller. This could be a public function in `conditions_util.go` and both controller could call it.

Reply:
Nice.  Included in next push.